src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+      // {"Equals", PredicateCondition::Equals},
+      // {"GreaterThan", PredicateCondition::GreaterThan}
+  };
+
+  const std::map<
+      std::string,
+      std::pair<std::function<std::shared_ptr<TableWrapper>(const int, const EncodingType, const OrderByMode)>,
+                std::function<std::shared_ptr<TableWrapper>(const int, const EncodingType, const OrderByMode)>>>
+      table_types{{"Int", std::make_pair(sorted_int_table, unsorted_int_table)},
+                  {"String", std::make_pair(std::bind(sorted_string_table, std::placeholders::_1, std::placeholders::_2,
+                                                      std::placeholders::_3, string_size),
+                                            std::bind(unsorted_string_table, std::placeholders::_1,
+                                                      std::placeholders::_2, std::placeholders::_3, string_size))}};
+
+  const std::map<std::string, OrderByMode> order_bys{
+      {"AscendingNullsFirst", OrderByMode::Ascending},
+      {"AscendingNullsLast", OrderByMode::AscendingNullsLast},
+      {"DescendingNullsFirst", OrderByMode::Descending},
+      {"DescendingNullsLast", OrderByMode::DescendingNullsLast},
+  };
+
+  const std::map<std::string, EncodingType> encoding_types{{"None", EncodingType::Unencoded},
+                                                           {"Dictionary", EncodingType::Dictionary}};
+
+  for (const auto& table_type : table_types) {
+    const auto data_type = table_type.first;
+    const auto sorted_table = table_type.second.first;
+    const auto unsorted_table = table_type.second.second;
+
+    for (const auto& predicate : predicates) {
+      const auto predicate_name = predicate.first;
+      const auto predicate_condition = predicate.second;
+
+      for (const auto& order_by : order_bys) {
+        const auto order_by_name = order_by.first;
+        const auto order_by_mode = order_by.second;
+
+        for (const auto& encoding : encoding_types) {
+          const auto encoding_name = encoding.first;
+          const auto encoding_type = encoding.second;
+
+          for (const auto selectivity : selectivities) {
+            benchmark::RegisterBenchmark(
+                ("BM_TableScanSorted/" + data_type + "/" + predicate_name + "/" + order_by_name + "/" + encoding_name +
+                 "/" + std::to_string(selectivity) + "/" + "Sorted")
+                    .c_str(),
+                BM_TableScanSorted, rows, selectivity, predicate_condition, encoding_type, order_by_mode, sorted_table);
+            benchmark::RegisterBenchmark(
+                ("BM_TableScanSorted/" + data_type + "/" + predicate_name + "/" + order_by_name + "/" + encoding_name +
+                 "/" + std::to_string(selectivity) + "/" + "Unsorted")
+                    .c_str(),
+                BM_TableScanSorted, rows, selectivity, predicate_condition, encoding_type, std::nullopt,
+                unsorted_table);
+          }
+        }
+      }
+    }
```

Comment:
This comment is not to be taken seriously but I just have quote this sentence from https://www.kernel.org/doc/Documentation/process/coding-style.rst

> ... if you need more than 3 levels of indentation, you're screwed anyway, and should fix your program.

Reply:
We think it's okay for this specific case.